scripts/linux/generic-rust/install.sh

```diff
@@ -49,7 +49,7 @@ function install_rust()
     # but it's simpler to just always `update` whatever toolchain it is. `update` installs the toolchain
     # if it hasn't already been installed, so this also works for pinned versions.
     source $HOME/.cargo/env
-    rustup update "$(< "$PROJECT_ROOT/rust-toolchain")"
+    rustup show active-toolchain
```

Comment:
Shouldn't this be `rustup update "$(rustup show active-toolchain)"` ?

Reply:
As far as I can tell, `rustup show active-toolchain` always synchronizes the tracked channel and updates if necessary.

Reply:
I just tried it on my machine where the default is nightly.

```sh
$ rustup show active-toolchain

nightly-x86_64-unknown-linux-gnu (default)

$ rustup update nightly

info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
716.3 KiB / 716.3 KiB (100 %) 703.5 KiB/s in  1s ETA:  0s
info: latest update on 2022-02-14, rust version 1.60.0-nightly (1e12aef3f 2022-02-13)
info: downloading component 'clippy'
...
```

So no, I don't think it implicitly updates.

Reply:
Interesting, will fix.

Reply:
Also, you can't directly do what I suggested, because

```sh
$ rustup update "$(rustup show active-toolchain)"

info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu (default)'
info: latest update on 2022-02-14, rust version 1.60.0-nightly (1e12aef3f 2022-02-13)
error: target 'x86_64-unknown-linux-gnu (default)' not found in channel.  Perhaps check https://doc.rust-lang.org/nightly/rustc/platform-support.html for available targets
```

So you'd need to `sed` out the trailing ` (default)`.

(You could do just `rustup update` but the disadvantage is that it updates all toolchains, not just the installed one. But maybe that's not a big deal.)

Reply:
Adding a `sed` is a small concession compared to downloading a completely new toolchain, so I will go with the `sed` route as you suggested.